### PR TITLE
Add auditor client capability (Node.js)

### PR DESCRIPTION
**Pull request by jnmt**
This patch set provides auditor client capabilities. CI testing will be passed after https://github.com/scalar-labs/scalardl-javascript-sdk-base/pull/57 is merged. Also, then I will update package-lock.json.

### CODE REVIEW

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Comment by jnmt** [comment]
@supl Currently we are using the latest docker instead of the installed one because its version in the machine image is too old as discussed before https://github.com/scalar-labs/scalardl-node-client-sdk/pull/35#discussion_r590926115. Now I found that we can use the latest installed one listed [here](https://circleci.com/docs/2.0/configuration-reference/#available-machine-images) if I use v2.1 of CircleCI. So, can we move to v2.1 and use the installed one? Do you have any concerns with v2.1? I think it's nice if we don't have to care about installing the docker. Thank you.

**Comment by supl** [comment]
> @supl Currently we are using the latest docker instead of the installed one because its version in the machine image is too old as discussed before [#35 (comment)](https://github.com/scalar-labs/scalardl-node-client-sdk/pull/35#discussion_r590926115). Now I found that we can use the latest installed one listed [here](https://circleci.com/docs/2.0/configuration-reference/#available-machine-images) if I use v2.1 of CircleCI. So, can we move to v2.1 and use the installed one? Do you have any concerns with v2.1? I think it's nice if we don't have to care about installing the docker. Thank you.

Thank you :)

We have a pipeline to install docker image and docker compose in CircleCI workflow.
It happens many times that the private key used in the install script is not registered in the CircleCI's ubuntu.

I also realize CircleCI provides new ubuntu with newer docker and docker compose when I tried to fix the CI failure in #41.

So, I agree with you.
We can try to use 2.1 and remove the docker install pipeline to save our life w

**Comment by supl** [comment]
@feeblefakie 
Sorry to put your out of the topic a bit.
Since thing has been changed, do we use another GitHub account for CircleCI's `GHCR_USERNAME` and `GHCR_PAT` context? Some official GitHub account like `scalarbot`.

File: .circleci/config.yml [hunk 1]
```diff
@@ -23,23 +18,13 @@ jobs:
...
-            curl -fsSL https://get.docker.com -o get-docker.sh
-            sudo sh get-docker.sh
-            sudo curl -L "https://github.com/docker/compose/releases/download/1.26.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
-            sudo chmod +x /usr/local/bin/docker-compose
-
       - run:
           name: checkout scalar-samples repository as it contains the script to spin up scalardl environment and easier for us to maintain the CI
           command: |
             git init
             git pull https://${GHCR_USERNAME}:${GHCR_PAT}@github.com/scalar-labs/scalar-samples.git
```
Comment (supl) [inline .circleci/config.yml:25]:
```suggestion
            git pull https://${GHCR_USERNAME}:${GHCR_PAT}@github.com/scalar-labs/scalardl-samples.git
```

**Review by supl — COMMENTED** [review]
Thank you so much !
It's LGTM.
Let's fix the integration test and wait for the base gets merged.

File: .circleci/config.yml (hunk 1 above)
Comment (jnmt) [reply .circleci/config.yml:25]: Thank you for the suggested change. Maybe CI fails due to this. But I'm curious about why `git pull` follows the change of the repo's name... In my environment, it seems to redirect `scalardl-samples` automatically.

**Comment by feeblefakie** [comment]
@supl 
> Since thing has been changed, do we use another GitHub account for CircleCI's GHCR_USERNAME and GHCR_PAT context? 

Sorry, I don't fully get what you mean. Can you clarify?

**Comment by supl** [comment]
This PR also change the Circle CI workflow.
Now it only tries to publish this package to NPM registry when a tag v[0-9]+.[0-9]+.[0-9]+ is pushed :)

**Comment by supl** [comment]
@jnmt 
Jun,
When I merged master into this branch, I got some conflicts at package-lock.json file, so I recreated it.
The package-lock.json is now changed to version 2 (lockfileVersion: 2) because my local npm is 7.
I think it should be ok with npm 6.
Just for your information.
Thank you :)

**Comment by feeblefakie** [comment]
@supl The CI fails. Can you take a look? (is it dependent on another PR?)

**Comment by supl** [comment]
> @supl The CI fails. Can you take a look? (is it dependent on another PR?)

Yes.
It failed on download SDK base (is not released yet)
But, I will get back to this PR if it has another fails besides download dependencies.

**Comment by jnmt** [comment]
@supl I updated README though it's the minimum level (web as well). PTAL. Thank you : )

**Comment by jnmt** [comment]
@supl @feeblefakie Auditor tests failed for both Node and Web. Let me check them.

**Comment by feeblefakie** [comment]
@supl Sorry for the interrupt, but can you fix the CI? 
Jun was trying but it seems like there is still an issue around the containers' setup. 
(just sleeping a s long time to wait for all the containers are ready wasn't worked well? > @jnmt )
I think it's about time to merge and release.

**Comment by supl** [comment]
> @supl Sorry for the interrupt, but can you fix the CI?
> Jun was trying but it seems like there is still an issue around the containers' setup.
> (just sleeping a s long time to wait for all the containers are ready wasn't worked well? > @jnmt )
> I think it's about time to merge and release.

No problem.
Let me do it :)

**Comment by jnmt** [comment]
> @supl Sorry for the interrupt, but can you fix the CI?
> Jun was trying but it seems like there is still an issue around the containers' setup.
> (just sleeping a s long time to wait for all the containers are ready wasn't worked well? > @jnmt )
> I think it's about time to merge and release.

@feeblefakie Sorry for not finishing yet. The long sleep time is not working well. I'm doubting that two concurrent schema loader takes longer time to finish than expect (cf. single loader in ledger-only case). So I was seeking how to execute the schema loader sequentially in scalar-samples with a good manner.
@supl I'm really happy if you help me out. Thanks.

**Comment by supl** [comment]
> > @supl Sorry for the interrupt, but can you fix the CI?
> > Jun was trying but it seems like there is still an issue around the containers' setup.
> > (just sleeping a s long time to wait for all the containers are ready wasn't worked well? > @jnmt )
> > I think it's about time to merge and release.
> 
> @feeblefakie Sorry for not finishing yet. The long sleep time is not working well. I'm doubting that two concurrent schema loader takes longer time to finish than expect (cf. single loader in ledger-only case). So I was seeking how to execute the schema loader sequentially in scalar-samples with a good manner.
> @supl I'm really happy if you help me out. Thanks.

Jun,
Thank you for your information.
It's very helpful.
I will try to run schema loaders in sequence.

**Comment by supl** [comment]
@feeblefakie @jnmt 
The CI issue is fixed.

**Comment by feeblefakie** [comment]
@supl Great! Thank you!

File: .circleci/config.yml [hunk 2]
```diff
@@ -72,13 +67,45 @@ jobs:
...
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up scalardl-ledger-schema-loader-cassandra
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up scalardl-auditor-schema-loader-cassandra
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-ledger
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-auditor
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d ledger-envoy
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d auditor-envoy
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-ledger-as-client
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-audior-as-client
+            sleep 5
+            docker restart scalardl-samples-scalar-ledger-1 # 念のために
```
Comment (feeblefakie) [inline .circleci/config.yml:97]: No Japanese please :)

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
Left a very minor comment.

File: .circleci/config.yml [hunk 3]
```diff
@@ -87,8 +87,16 @@ jobs:
...
             docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up scalardl-ledger-schema-loader-cassandra
             docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up scalardl-auditor-schema-loader-cassandra
-            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-ledger scalar-auditor ledger-envoy auditor-envoy
-            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-ledger-as-client scalar-audior-as-client
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-ledger
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-auditor
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d ledger-envoy
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d auditor-envoy
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-ledger-as-client
+            docker-compose -f docker-compose.yml -f docker-compose-auditor.yml up -d scalar-audior-as-client
```
Comment (jnmt) [inline .circleci/config.yml]: Wow, the way to execute sequentially is simple and good enough : )

File: .circleci/config.yml [hunk 4]
```diff
@@ -87,8 +87,16 @@ jobs:
...
+            sleep 5
+            docker restart scalardl-samples-scalar-ledger-1 # 念のために
+            docker restart scalardl-samples-scalar-auditor-1 # 念のために
```
Comment (jnmt) [inline .circleci/config.yml:98]: Should be "# just in case" ?? (Web might as well)

**Review by jnmt — COMMENTED** [review]
@supl Thank you so much! If possible, can we remove Japanese?

File: .circleci/config.yml (hunk 4 above)
Comment (supl) [reply .circleci/config.yml:98]: No problem.
Let me change it.
(Actually maybe the comments are not necessary w)

**Review by supl — APPROVED** [review]
It's an LGTM.
Thank you :)

**Comment by feeblefakie** [comment]
@jnmt @supl Thank you, guys!

**Pull request merged**